From foo@baz Mon Jan  5 15:45:21 PST 2009
Date: Mon, 05 Jan 2009 15:45:21 -0800
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: epl: fix netdev->priv b0rkage

From: Greg Kroah-Hartman <gregkh@suse.de>

netdev->priv is now gone, use netdev_priv() instead.
This fixes the build error in the network driver within the epl stack.

Cc: Daniel Krueger <daniel.krueger@systec-electronic.com>
Cc: Ronald Sieber <Ronald.Sieber@systec-electronic.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/epl/VirtualEthernetLinux.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

--- a/drivers/staging/epl/VirtualEthernetLinux.c
+++ b/drivers/staging/epl/VirtualEthernetLinux.c
@@ -166,7 +166,7 @@ static int VEthOpen(struct net_device *p
 	tEplKernel Ret = kEplSuccessful;
 
 	//open the device
-//    struct net_device_stats* pStats = (struct net_device_stats*)pNetDevice_p->priv;
+//	struct net_device_stats* pStats = netdev_priv(pNetDevice_p);
 
 	//start the interface queue for the network subsystem
 	netif_start_queue(pNetDevice_p);
@@ -199,8 +199,7 @@ static int VEthXmit(struct sk_buff *pSkb
 	tEplFrameInfo FrameInfo;
 
 	//transmit function
-	struct net_device_stats *pStats =
-	    (struct net_device_stats *)pNetDevice_p->priv;
+	struct net_device_stats *pStats = netdev_priv(pNetDevice_p);
 
 	//save timestemp
 	pNetDevice_p->trans_start = jiffies;
@@ -233,7 +232,7 @@ static struct net_device_stats *VEthGetS
 {
 	EPL_DBGLVL_VETH_TRACE0("VEthGetStats\n");
 
-	return (struct net_device_stats *)pNetDevice_p->priv;
+	return netdev_priv(pNetDevice_p);
 }
 
 static void VEthTimeout(struct net_device *pNetDevice_p)
@@ -250,8 +249,7 @@ static tEplKernel VEthRecvFrame(tEplFram
 {
 	tEplKernel Ret = kEplSuccessful;
 	struct net_device *pNetDevice = pVEthNetDevice_g;
-	struct net_device_stats *pStats =
-	    (struct net_device_stats *)pNetDevice->priv;
+	struct net_device_stats *pStats = netdev_priv(pNetDevice);
 	struct sk_buff *pSkb;
 
 	EPL_DBGLVL_VETH_TRACE1("VEthRecvFrame: FrameSize=%u\n",
